File: doc-examples/example-java/src/test/java/example/ReaderExample.java
package example;

import io.micronaut.pulsar.annotation.PulsarReader;
import jakarta.inject.Singleton;
import org.apache.pulsar.client.api.Message;
import org.apache.pulsar.client.api.Reader;

import java.util.concurrent.CompletableFuture;


@Singleton
public class ReaderExample {

    @PulsarReader(value = "persistent://public/default/messages", readerName = "simple-j-reader") // <1>
    private Reader<String> reader; // <2>

    public CompletableFuture<Message<String>> readNext() { // <3>
        return reader.readNextAsync(); // <4>
    }
}

File: doc-examples/example-python/src/main/python/example/ReaderExample.py
from typing import Annotated

from jakarta.inject import Inject, Singleton
from java.util.concurrent import CompletableFuture
from micronaut.pulsar.annotation import PulsarReader
from org.apache.pulsar.client.api import Message, Reader


@Singleton
class ReaderExample:

    reader: Annotated[Reader[str], Inject, PulsarReader("persistent://public/default/messages", readerName="simple-py-reader")]  # <1> <2>

    def read_next(self) -> CompletableFuture[Message[str]]:  # <3>
        return self.reader.readNextAsync()  # <4>
